DCOM oaut.ExceptionInfo panics during unmarshal

I'm using this library to make a golang implementation of [Impacket's dcomexec.py](https://github.com/fortra/impacket/blob/master/examples/dcomexec.py), and I ran into an issue when running against a Windows 11 machine.
```
panic: interface conversion: interface {} is **oaut.Variant, not **oaut.String

goroutine 67 [running]:
github.com/oiweiwei/go-msrpc/msrpc/dcom/oaut.(*ExceptionInfo).UnmarshalNDR.func2({0x832b00?, 0xc000122730?})
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/msrpc/dcom/oaut/oaut.go:7412 +0x65
github.com/oiweiwei/go-msrpc/ndr.(*ndr20).ReadPointer(0xc0002b2b60, {0x832bc0, 0xc000326a48}, 0xc0003671a0, {0xc0003671b0, 0x1, 0x40?})
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/ndr/ndr20.go:490 +0x22a
github.com/oiweiwei/go-msrpc/msrpc/dcom/oaut.(*ExceptionInfo).UnmarshalNDR(0xc000326a40, {0x832b00?, 0xc000122730?}, {0x9b8ae0, 0xc0002b2b60})
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/msrpc/dcom/oaut/oaut.go:7413 +0x182
github.com/oiweiwei/go-msrpc/msrpc/dcom/oaut/idispatch/v0.(*xxx_InvokeOperation).UnmarshalNDRResponse(0xc000122700, {0x9b5568, 0xc0002946f0}, {0x9b8ae0, 0xc0002b2b60})
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/msrpc/dcom/oaut/idispatch/v0/v0.go:1773 +0x269
github.com/oiweiwei/go-msrpc/dcerpc.NewBody.func1()
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/dcerpc/packet.go:137 +0x112
created by github.com/oiweiwei/go-msrpc/dcerpc.NewBody in goroutine 1
        /home/bryan/go/pkg/mod/github.com/oiweiwei/go-msrpc@v1.2.1/dcerpc/packet.go:129 +0x1f8
```

The code looks something like this:
```go
// ...
    this := &dcom.ORPCThis{
      Version:    ComVersion,
      Flags:      0,
      CID:        &RandCid,
      Extensions: nil,
    }

    gr, err := dc.GetIDsOfNames(ctx, &idispatch.GetIDsOfNamesRequest{
      This:     this,
      IID:      NullIID,
      Names:    []string{"Document\x00"},
      LocaleID: 0,
    })
    if err != nil {
      return err
    }
    if len(gr.DispatchID) < 1 {
      return fmt.Errorf("no dispatch ID found")
    }
    mod.log.Info().Int("matches", len(gr.DispatchID)).Msg("Document")

    // This call causes the panic
    pr, err := dc.Invoke(ctx, &idispatch.InvokeRequest{
      This:             this,
      DispatchIDMember: gr.DispatchID[0],
      IID:              NullIID,
      Flags:            2, // PropertyGet
      LocaleID:         0x409, // English
      DispatchParams:   &oaut.DispatchParams{},
    })
// ...
```

If any more info is needed, please let me know.

Edit: I just realized I've included a panic from a modified `oaut.go`. This issue is persistent directly from the main branch as well, so ignore that.

Edit 2: changed panic trace to original

--- a/msrpc/dnsp/record/rr.go
+++ b/msrpc/dnsp/record/rr.go
@@ -345,7 +345,11 @@ func NewRecordFromRR(rr dns.RR) (*Record, error) {
 	case *ms_dns.WINS:
 		ips := make([][]byte, len(rr.WINSServers))
 		for i := range ips {
-			ips[i] = rr.WINSServers[i]
+			ip := rr.WINSServers[i]
+			if ip4 := ip.To4(); ip4 != nil {
+				ip = ip4
+			}
+			ips[i] = ip
 		}
 		val.Value = &Record_WINS{RecordWINS: &RecordWINS{
 			MappingFlag:   rr.MappingFlag,
@@ -680,7 +684,7 @@ func (r *Record) RR() dns.RR {
 		for i := range ips {
 			ips[i] = rr.WINSServers[i]
 			if ip4 := ips[i].To4(); ip4 != nil {
-				ips[i] = ips[i].To4()
+				ips[i] = ip4
 			}
 		}
 		return &ms_dns.WINS{

--- a/ndr/buffer.go
+++ b/ndr/buffer.go
@@ -364,8 +364,14 @@ func (c *WaitChunk) Float() math.FloatFormat {
 }
 
 // Bytes function returns the bytes written/not-yet-read for the
-// chunk.
-func (c *WaitChunk) Bytes() []byte { return nil /* panic("wait_chunk: bytes not supported") */ }
+// chunk. (unsafe, use for debug only)
+func (c *WaitChunk) Bytes() []byte {
+	if c.bAndFormat != nil {
+		return c.bAndFormat.b
+	}
+	return nil
+	/* panic("wait_chunk: bytes not supported") */
+}
 
 func (c *WaitChunk) ReadRepresentation(drep *DataRepresentation) error {
 	panic("wait_chunk: read_representation not supported")

--- a/ndr/ndr20.go
+++ b/ndr/ndr20.go
@@ -496,6 +496,11 @@ func (w *ndr20) ReadPointer(ptr Pointer, setter func(interface{}), mrs ...Unmars
 		return nil
 	}
 
+	if pptr == 0x72657355 /* 'User' */ {
+		w.rdeferred = append(w.rdeferred, mrs...)
+		return nil
+	}
+
 	if ptr, ok := w.ptrs[uint64(pptr)]; ok {
 		setter(interface{}(ptr))
 		return nil

